scripts/check_python_module

```diff
@@ -36,8 +37,16 @@ except ImportError:
 #
 # Check version compatibility
 #
-needver = list(map(int, version.split('.')))
-havever = list(map(int, module.__version__.split('.'))) + [0] * len(needver)
+astropy_version = module.__version__
+# Remove any trailing characters from astropy version numbers after the
+# first character that is neither a digit nor a ".".
+# This does mean that 4.0.0.rc1 would become 4.0.0. so we have to clean that
+astropy_version = re.sub(r"[^.\d].*$", "", astropy_version)
+# Trailing "."
+astropy_version = re.sub(r"\.$", "", astropy_version)
+
+needver = [int(d) for d in version.split(".")]
+havever = [int(d) for d in astropy_version.split('.')] + [0] * len(needver)
```

Comment:
It feels like it would be more straightforward to do this with a single call to `re.match` rather than multiple calls to `re.sub`. Something like:
```python
havever = [int(d) for d in re.match(r"(\d+)\.(\d+)\.(\d+)", astropy_version).groups())]
```
perhaps?

Also, it might be helpful to add a comment stating explicitly here that we are regarding an RC as equivalent to a full release.

Reply:
Yes, although I see the actual version string is "4.0rc1" so that would not match because there are not 3 numbers.

Reply:
Also also, the `len(needver)` stuff is super confusing — I know you're not adding it here, but could you make it something like `[0] * (len(needver) - len(havever)`), which I think is what it actually intends.

Reply:
I was looking at the len and realized that because of the zip later it really doesn't matter that a few extra 0s have turned up.

Reply:
Right — the logic is correct, but the code is unnecessarily obtuse.